src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -3084,27 +3182,114 @@ def pv_capital_cost_constraint(c):
 
             @b.Constraint()
             def pv_capital_cost_constraint(c):
+                # no expenditure period, so cash flow occurs at t=0 (project year)
                 # PV_Capital_Cost = - CAPEX
 
                 return c.pv_capital_cost == -pyunits.convert(
-                    c.CAPEX
-                    * series_present_worth_factor(
+                    c.CAPEX,
+                    to_units=c.cost_units,
+                )
+
+        if b.config.debt_expression is None:
+
+            @b.Constraint()
+            def loan_debt_constraint(c):
+                # Debt  = %debt_charge_of_CAPEX * CAPEX
+
+                return c.loan_debt == pyunits.convert(
+                    pyunits.convert(
+                        c.debt_percentage_of_CAPEX, to_units=pyunits.dimensionless
+                    )
+                    * c.CAPEX,
+                    to_units=c.cost_units,
+                )
+
+        else:
+
+            b.loan_debt = Reference(b.config.debt_expression)
+
+        @b.Constraint()
+        def loan_annual_payment_constraint(c):
+            # Annual Loan Payment = Debt * A/P(%interest, 0, loan_length) = Debt / P/A(%interest, 0, loan_length)
+            # we assume that the loan payment series starts in year 1, so we don't need to account for a delay
+
+            if c.config.debt_expression is None:
+
+                return c.loan_annual_payment == pyunits.convert(
+                    c.loan_debt
+                    / series_present_worth_factor(
                         pyunits.convert(
-                            c.capital_loan_interest_percentage,
+                            c.capital_loan_interest_percentage,  # payment value is discounted by loan interest rate
                             to_units=pyunits.dimensionless,
                         ),
+                        0,  # payments adjusted to present value are constant
+                        c.capital_loan_repayment_period / pyunits.year,
+                    ),
+                    to_units=c.cost_units,
+                )
+
+            else:
+
+                return c.loan_annual_payment == pyunits.convert(
+                    c.loan_debt[None]
+                    / series_present_worth_factor(
                         pyunits.convert(
-                            c.capital_escalation_percentage,
+                            c.capital_loan_interest_percentage,  # payment value is discounted by loan interest rate
                             to_units=pyunits.dimensionless,
                         ),
-                        0,
-                    ),  # formula gives P/A (r, g, 0) = 1
+                        0,  # payments adjusted to present value are constant
+                        c.capital_loan_repayment_period / pyunits.year,
+                    ),
+                    to_units=c.cost_units,
+                )
+
+        @b.Constraint()
+        def pv_loan_interest_constraint(c):
```

Comment:
Ultimately I think the present value of the loan payments + interest should be the following:
series present worth factor (same as used throughout)* interest rate* loan principal/ (1- (1+ interest rate)^(-n)). Does this come out to be the same thing?

Reply:
Our two formulations are similar. Mine simplifies to

$\frac{P_A (r, i, N)}{P_A (i, 0, N)} * Debt$

while yours simplifies to

$\frac{P_A (r, i, N)}{P_A (i, 0, N)} * Debt$

where $P_A$ is the series present worth factor, $r$ is the discount rate, $i$ is the interest rate, $N$ is the loan length, and $Debt$ is the loan principal. So I have the annual payment series growing over time by the interest, and in yours it doesn't grow over time.

Reply:
Ok thank you brandon for simplifying the expression! I am not familiar with including escalation/ growth for a debt payment (as long as the interest rate stays constant).

Reply:
I thought about this some more, and I think you're correct that the annual payments shouldn't be escalating over time. There are two possible formulas we could use to get the total sum of annual payments over the repayment period:

1. Your suggestion, $TotalLoanPlusInterestPayments = \frac{P_A (r, 0, N)}{P_A (i, 0, N)} * Debt = P_A (r, 0, N) * Annual_Payment_Amount$ which assumes that the calculated annual payments are absolute amounts that must be adjusted to present day dollars. The problem here is that when the interest rate is less than the discount rate, we get a positive present value - so we're making money by taking out the loan. When using the WACC as the discount rate for a 10 year loan, the total interest is only 1% of the principal, which is very small compared to the interest rate of 6%.

2. $TotalLoanPlusInterestPayments = \frac{N}{P_A (i, 0, N)} * Debt = N * Annual_Payment_Amount$ which assumes that the calculated annual payments are already in present day dollars are we can just multiply by the number of payments. This will always be greater than the original Debt, so there is always going to be a negative present value - a cost of taking out the loan. For a 10 year loan at 6% interest, the total interest is 35% of the principal, which seems more reasonable to me since the unpaid debt balance is accumulating more interest each year.

Reply:
Hi Brandon- thanks so much for looking into this further. I agree with you that the second approach makes more sense. I am good with going ahead with that!

Reply:
I implemented the formulation discussed in recent meetings and offline and consolidated into a single expression.